core/src/main/java/org/apache/iceberg/PartitionsTable.java

```diff
@@ -164,17 +165,20 @@ Partition get(StructLike key) {
   }
 
   static class Partition {
+    private int specId;
     private final StructLike key;
     private long recordCount;
     private int fileCount;
 
     Partition(StructLike key) {
+      this.specId = 0;
```

Comment:
I'm wondering one thing , didn't test it myself.  

If we have two partitions with different spec id but different key, it will show up with a random one.  Could this case happen (ie, adding and removing a field in partition spec should give a new spec id but can still produce the same partition value?)?  If so, we may need to add spec-id to key?

Reply:
I see, yea you are right, seems Iceberg re-uses the spec id if we revert the spec to a former one.